crates/build/src/wasm_opt.rs

```diff
@@ -68,9 +69,14 @@ impl WasmOptHandler {
         );
 
         OptimizationOptions::from(self.optimization_level)
-            // Since rustc 1.70 `SignExt` can't be disabled anymore. Hence we have to allow it.
             .mvp_features_only()
+            // Since rustc 1.70 `SignExt` can't be disabled anymore. Hence we have to allow it,
+            // in order that the Wasm binary containing these instructions can be loaded.
             .enable_feature(Feature::SignExt)
+            // This pass will then remove any `signext` instructions in order that the resulting
+            // Wasm binary is compatible with older versions of `pallet-contracts` which do not
+            // support the `signext` instruction.
+            .add_pass(Pass::SignextLowering)
```

Comment:
Since https://github.com/paritytech/substrate/pull/14565 we do allow sign_extension in smart contracts on the pallet side. Shouldn't we make this pass optional for the cargo-contract so that the users of the newer versions of pallet-contracts could still opt in to have signext instructions in their contracts?

Reply:
I am aware, and I did consider making this optional. I decided against it because removing them by default means that both old and new versions of `pallet_contracts` will be supported with this, so will make users and our lives easier if the resulting wasm is compatible with as wide a range of target nodes as possible.

I assume the majority of node implementations currently are using an older version of `pallet_contracts` which does not support `signext`.

Reply:
I see. Could adding smth like `--allow-signext` to the `cargo-contract contract build` so that `wasm_opt` runs without the pass, be an option then?